### ISSUE TITLE
[notifications] settings not applied

### ISSUE THREAD

**Issue by despecial** [downthemall/downthemall#148]
**Desktop (please complete the following information):**
- Mac OS Mojave
- Firefox 69.0.2

**Describe the bug**
Preferences set to:
- Show a notification when the queue finished to false
- Play sounds to false

Still get sound plus popup window when queue has finished 

**To Reproduce**
Got to preferences and disable mentioned settings above

**Expected behavior**
No sound, no popup dialogue

**Screenshots**
<img width="848" alt="Bildschirmfoto 2019-10-06 um 14 33 55" src="https://user-images.githubusercontent.com/4135109/66269262-daef9580-e846-11e9-8624-8734a8889777.png">

**Comment by nmaier** [comment]
This works for me...

**Comment by despecial** [comment]
Thats nice but not here. I get both the popup and the sound.

**Comment by nmaier** [comment]
Do the settings actually stay the same (unchecked) after you a) reload the tab b) reload the extension (disable and enable again) and c) restart the browser?

**Comment by despecial** [comment]
a) reload tab
yes, same settings

b) enable/disable extension
yes, stay the same

c) restart browser
still, all settings stay the same

Test download queue: popup and sound playing

**Comment by nmaier** [comment]
It's hard to fix this without being able to reproduce this, of course.

But I think I found an issue that could in theory cause this in certain browser configurations...

**Comment by nmaier** [comment]
I committed ^ something that may or may not fix the issue. At least it shouldn't break anything else.

It would be great if you could comment on whether this resolves it for you:

- Either wait for the 4.2 release (sometime middle-ish of next week)
- Or if you feel advantageous and up to it you could try to run the development version from a git clone, following the steps in https://github.com/downthemall/downthemall#development

**Comment by despecial** [comment]
Interesting. When I check the xpi file in my library I dont even see a libs folder.
This is what I see.
<img width="162" alt="Bildschirmfoto 2019-10-06 um 22 51 24" src="https://user-images.githubusercontent.com/4135109/66275737-ed3ff280-e88b-11e9-81fe-a3084b97487a.png">

As for the development version, I will check :) Thanks!

**Comment by nmaier** [comment]
yes, the files in libs get transpiled from typescript into javascript that a browser actually understands, bundles up and placed into a few large javascript files in `bundles`

**Comment by despecial** [comment]
I see, just wanted to edit the current xpi content and check if the change works.
But I guess the yarn way is more effective

**Comment by nmaier** [comment]
And you cannot just modify the xpi anyway, as the browser will check for a valid signature from mozilla and any changes would turn the signature invalid,

**Comment by despecial** [comment]
hanging on the build process. I dont have a global python3 env. Damn

**Comment by nmaier** [comment]
You don't need to build at all if you just `yarn`, `yarn watch` + `yarn webext`

**Comment by despecial** [comment]
Just updated to the new version. Set the options again, restartet the browser.
Still showing a popup notification plus sound. :(

**Comment by despecial** [comment]
Thanks for the effort though. Wondering if anyone else has this problem.
The settings show the correct values for the options but they seem not to be applied by the script.

**Comment by nmaier** [comment]
Well, the one other bug report I marked duplicate of this one. Other than that, nobody reported the issue, not here or not by emailing me.

**Comment by msv6r** [comment]
I have also encountered the problem described by @despecial and @menma1234 running on Ubuntu 20.04.

**Comment by kpedro88** [comment]
This still happens for me (FF 117 on Win10). If I go to preferences, turn "Show a notification when the queue finishes downloading" back *on*, close the tab, reopen the preferences tab and deselect that option again, it is properly applied. However, upon restarting the browser, the setting is lost.
